subsys/mgmt/mcumgr/smp_bt.c

```diff
@@ -53,6 +54,9 @@ static ssize_t smp_bt_chr_write(struct bt_conn *conn,
 	struct net_buf *nb;
 
 	nb = mcumgr_buf_alloc();
+	if (!nb) {
+		return -MGMT_ERR_ENOMEM;
+	}
```

Comment:
I don't think this is right. We should return some error code that is understandable by BT GATT stack.

Reply:
``` suggestion
	if (!nb) {
		return BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
	}
```
?

Reply:
It is definitely better. Would be great if somebody from BT could verify that this is the right error code to use.

Reply:
@Thalley suggestion on Slack was:
> Given the context I would probably return `BT_ATT_ERR_INSUFFICIENT_RESOURCES`

Reply:
Would update it, thanks!